javaslang/src/test/java/javaslang/collection/AbstractTraversableTest.java

```diff
@@ -1326,34 +1326,45 @@ public void shouldReplaceAllElementsOfNonNilUsingCurrNew() {
 
     @Test
     public void shouldRetainAllElementsFromNil() {
+        Traversable<Object> src = empty();
+        Traversable<Object> expected = src;
+        Traversable<Object> actual = src.retainAll(of(1, 2, 3));
         if (useIsEqualToInsteadOfIsSameAs()) {
-            assertThat(empty().retainAll(of(1, 2, 3))).isEqualTo(empty());
+            assertThat(actual).isEqualTo(expected);
         } else {
-            assertThat(empty().retainAll(of(1, 2, 3))).isSameAs(empty());
+            assertThat(actual).isSameAs(expected);
         }
     }
 
     @Test
     public void shouldRetainAllExistingElementsFromNonNil() {
-        assertThat(of(1, 2, 3, 1, 2, 3).retainAll(of(1, 2))).isEqualTo(of(1, 2, 1, 2));
+        Traversable<Integer> src = of(1, 2, 3, 2, 1, 3);
+        Traversable<Integer> expected = of(1, 2, 2, 1);
+        Traversable<Integer> actual = src.retainAll(of(1, 2));
+        assertThat(actual).isEqualTo(expected);
     }
 
     @Test
     public void shouldRetainAllElementsFromNonNil() {
+        Traversable<Integer> src = of(1, 2, 1, 2, 2);
+        Traversable<Integer> expected = of(1, 2, 1, 2, 2);
+        Traversable<Integer> actual = src.retainAll(of(1, 2));
         if (useIsEqualToInsteadOfIsSameAs()) {
-            assertThat(of(1, 2, 1, 2, 2).retainAll(of(1, 2))).isEqualTo(of(1, 2, 1, 2, 2));
+            assertThat(actual).isEqualTo(expected);
         } else {
-            final Traversable<Integer> src = of(1, 2, 1, 2, 2);
-            assertThat(src.retainAll(of(1, 2))).isSameAs(src);
+            assertThat(actual).isSameAs(src);
```

Comment:
`isSameAs(src)` and not `isSameAs(expected)` since not all `Traversable`s are consumable multiple times